source/connection-monitoring-and-pooling/tests/cmap-format/pool-checkout-returned-connection-maxConnecting.json

```diff
@@ -70,11 +71,6 @@
     {
       "name": "checkIn",
       "connection": "conn0"
-    },
-    {
-      "name": "waitForEvent",
-      "event": "ConnectionCheckedOut",
-      "count": 4
     }
```

Comment:
I think, we need another `wait` after `checkIn`. The `wait` before `checkIn` ensures (not really, but good enough) that all activities are blocked (waiting for 4 `ConnectionCheckOutStarted` events is not enough for that): two threads are blocked for 750 ms by the failpoint, and one is blocked on `maxConnecting`. The proposed `wait` after `checkIn` ensures that there is enough time for one of `thread1`, `thread2`, `thread3` to checkout `conn0` that has just been checked in, and deliver the corresponding `ConnectionCheckedOut` event.

When I run the test as it currently is in the PR, it fails for me right away, because the last `ConnectionCheckedOut` fails to be delivered quickly enough. The proposed `wait` fixes that.

Reply:
Would waiting for the 2nd `ConnectionCheckedOut` events work also? That might be more reliable than adding a sleep.

E.g. add this 
```json
{
  "name": "waitForEvent",
  "event": "ConnectionCheckedOut",
  "count": 2
}
```

Reply:
The main idea of the fix applied in this PR, is that we to do all the work and checks while the two threads blocked by the failpoint keep holding all the `maxConnecting` permits. We know that they are blocked for at least 750 ms, so us waiting twice for at least 100 ms, should be OK. But if we start waiting for any events instead, we can no longer be sure that the above will hold. Thus, I don't think that is a viable approach.

Reply:
Ok, will you make the change?

Reply:
Yes, updated!